Release 1.0.0

KochTobi opened the pull request.

Aline-9 committed:
Feature/new subcommand (#126)

* Added subcommands for download and status

* change of picocli structure

* adding logout method and other smaller changes

* small change I forgot to add

* - update Readme
- remove unused options
- remove logout function again

* fixing the suffix option for the case that there is no file available for the given suffix

* Avoid exception swallowing

* Correctly filter download.

Correctly filters the downloaded files. Skips identifiers for which no files are present for the given filter rule. This needed to be done in order to avoid an influx of "Nothing to download" messages being presented to the user.

Co-authored-by: Tobias Koch <tobias.koch@qbic.uni-tuebingen.de>

KochTobi committed:
Add distribution management

KochTobi committed:
merge master

sven1103 commented:
Dear @Aline-9 , thanks a lot for this great contribution, I really enjoyed the two commands and especially the new status command will cover a lot of daily use cases for our platform users.

I have only two requests:

1. the `status` command name is not intuitive imo, I was asking myself: status of what? I think a name like `list`, `show`, `display` or `info` is closer to what the command actually does
2. the output of the `status` command needs some discussion

To point 2:

I take this example to make my point:

```
Number of datasets found for identifier Q****259A4 : 5
        Dataset 20220913152223334-969414 (NGS02Q****259A4)
                2013-05-07 03:57:15     20069a072_03.metadata   506 Bytes
                2013-05-07 03:57:15     20069a072_02_S1_L004_R2_001.fastq.gz    4525250895 Bytes
                2013-05-07 03:57:15     20069a072_02_S1_L004_R1_001.fastq.gz    4294340311 Bytes
                2013-05-07 03:57:15     20069a072_02_S1_L003_R2_001.fastq.gz    4484922384 Bytes

```

- Imo there is no value added to show the openBIS dataset ID, since it is a techncial id we dont use for communication. The important ID is the QBiC dataset id which is for example `NGS02Q....`

- I think the output format can be adjusted to improve parsing and also support display of a more nested query, like from a tissue sample id input (one level above the isolate sample id!):

```
Number of datasets found for identifier Q****259A4 : 5

# Dataset NGS02Q****259A4
# Source Q****259A4
# Registration 2013-05-07 03:57:15 
# Size xxxxx Bytes
20069a072_03.metadata   506 Bytes
20069a072_02_S1_L004_R2_001.fastq.gz    4525250895 Bytes
20069a072_02_S1_L004_R1_001.fastq.gz    4294340311 Bytes
20069a072_02_S1_L003_R2_001.fastq.gz    4484922384 Bytes
20069a072_02_S1_L003_R1_001.fastq.gz    4252961854 Bytes
20069a072_02_S1_L002_R2_001.fastq.gz    4410209477 Bytes
20069a072_02_S1_L002_R1_001.fastq.gz    4204281281 Bytes
20069a072_02_S1_L001_R2_001.fastq.gz    4409501161 Bytes
20069a072_02_S1_L001_R1_001.fastq.gz    4209466799 Bytes

# Dataset VC02Q****259A4
# Source Q****259A4
# Registration 2013-05-07 03:57:15 
# Size xxxxx Bytes
20069a072_03_var_annotated.vcf.gz       1342237539 Bytes
20069a072_03.metadata   586 Bytes

# Dataset NGS03****259A4
# Source Q****259A4
# Registration 2013-05-07 03:57:15 
# Size xxxxx Bytes
20069a072_03_S34_L004_R2_001.fastq.gz   3989522421 Bytes
20069a072_03_S34_L004_R1_001.fastq.gz   3742815787 Bytes
20069a072_03_S34_L003_R2_001.fastq.gz   3953472353 Bytes
20069a072_03_S34_L003_R1_001.fastq.gz   3723386625 Bytes
20069a072_03_S34_L002_R2_001.fastq.gz   3918273071 Bytes

...
```

The time display for every file is redundant, since they are all registered in the same operational step during ETL, thus they will be always the same within one dataset.

I would also remove the indentation, since command line tools output should be easy to process for downstream applications. The above suggestion provides a solution for humans and machines to iterate though datasets.

- Can you please provide an option for the `status` command to display human-readable file-sizes? I recommend to always display three digits plus one fractional digit, for example 1200 Bytes become 1.2 KB or 123 600 Bytes become 123.6 KB

Best, Sven

KochTobi committed:
remove deprecated method

KochTobi committed:
Include changes for release 1.0.0 (#131)

* - changed name of status subcommand --> display
- different output format for display subcommand
- human-readable file sizes
- TODO: change Readme

* fixed error message for creating parent directory during download
--> no error if (part of the) directory already exists

* Rename from display to list

* Format time and date in ISO-8601 including time zone

* Cut nano seconds and print UTC time

* print sample information (BREAKS IMPLEMENTATION)

* Find Q_TEST_SAMPLE analyte

* migrating type to use sample

* Add file size

* Consistently use Kilobyte, Megabyte, Gigabyte, Terabyte

Switches from KiB, MiB, GiB and TiB to KB, MB, GB and TB.

* Format file size output

* Remove unused code

* remove old unit display

* simplify best unit type choice

* renaming status subcommand to list

* format file sizes respect minwidth

* sort files in lexicographic order

* extract method

* Update README.md

Co-authored-by: Sven F. <9976560+sven1103@users.noreply.github.com>

Co-authored-by: Aline-9 <aline.breitinger@student.uni-tuebingen.de>
Co-authored-by: Sven F. <9976560+sven1103@users.noreply.github.com>
